spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureCloudConfigWatch.java

```diff
@@ -111,16 +115,22 @@ public void refreshConfigurations() {
      */
     private Boolean refresh(ConfigStore store, String storeSuffix, String watchedKeyNames) {
         String storeNameWithSuffix = store.getName() + storeSuffix;
-        QueryOptions options = new QueryOptions().withKeyNames(watchedKeyNames)
-                .withLabels(store.getLabels()).withFields(QueryField.ETAG).withRange(0, 0);
-
-        List<KeyValueItem> keyValueItems = configOperations.getRevisions(store.getName(), options);
+        SettingSelector settingSelector = new SettingSelector().setKeys(watchedKeyNames).setLabels(store.getLabels())
+                .setRange(new Range(0, 0));
+
+        List<ConfigurationSetting> items = null;
+        try {
+            items = clientStore.listSettingRevisons(settingSelector, store.getName());
+        } catch (ServerException e) {
```

Comment:
I am assuming the underlying protocol is http?
If that's the case, it's strange to use one rmi exception.

Reply:
This shouldn't be there. This exception isn't thrown, removing.